elitzur-avro/src/main/scala/com/spotify/elitzur/converters/avro/dynamic/DynamicAccessorValidator.scala

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2021 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.spotify.elitzur.converters.avro.dynamic
+
+import com.spotify.elitzur.MetricsReporter
+import com.spotify.elitzur.converters.avro.dynamic.dsl.AvroObjMapper
+import com.spotify.elitzur.validators.{DynamicRecordValidator, Unvalidated, Validator}
+import org.apache.avro.Schema
+import org.apache.avro.generic.GenericRecord
+
+import scala.util.{Failure, Success, Try}
+
+class DynamicAccessorValidator(
+  recordAccessorWithValidator: Array[(String, DynamicValidationCompanion)],
```

Comment:
This is the main assumption of this PR. There will be some code further upstream that can create this input: i.e., that code would know how to:
1) Given a string input, knows how to get this `DynamicValidationCompanion`. This will most likely be a map that lives in the [elitzur-validation-types](https://ghe.spotify.net/datainfra/elitzur-validation-types) repo.
2) It will know how to create this Array that will get passed as an input to this DynamicAccessorValidator class

Reply:
This seems like a fair assumption to me, we already have code that works like this in the internal repo.